web/src/engine/websites/DayComics.ts

```diff
@@ -0,0 +1,119 @@
+import { Tags } from '../Tags';
+import icon from './DayComics.webp';
+import { Chapter, DecoratableMangaScraper, Manga, type MangaPlugin } from '../providers/MangaPlugin';
+import * as Common from './decorators/Common';
+import { FetchJSON, FetchWindowScript } from '../platform/FetchProvider';
+
+const pageScript = `
+    new Promise ( resolve =>  resolve([...document.querySelectorAll('div#comicContent div.imgSubWrapper img')].map(image => image.dataset.src)));
+`;
+
+type APIResult<T> = {
+    data: T
+}
+
+type APIComicDetails = {
+    comic: APIComic,
+    episode : APIChapter[]
+}
+
+type APIComic = {
+    comicId: number,
+    language: string,
+    information: {
+        title: string
+    }
+}
+
+type APIComicList = {
+    new?: APIComic[],
+    binge?: APIComic[],
+    montly?: APIComic[]
+}
+
+type APIChapter = {
+    episodeId: number,
+    information: {
+        title: string,
+        subtitle : string
+    }
+}
+
+type MatureToken = {
+    token: string
+}
+
+@Common.PagesSinglePageJS(pageScript, 1500)
+@Common.ImageAjax()
+
+export default class extends DecoratableMangaScraper {
+
+    private readonly apiUrl = 'https://api.daycomics.com/api/';
+    public constructor() {
+        super('daycomics', `DayComics`, 'https://daycomics.com', Tags.Language.English, Tags.Media.Manhwa, Tags.Source.Official);
+    }
+
+    public override get Icon() {
+        return icon;
+    }
+
+    public override async Initialize(): Promise<void> {
+        //Set +18 cookie for not logged person. Logged people will have to take care of that themselves (-change account settings )
```

Comment:
How will the user get this information?
Shouldn't there be an error for the user when accessing such media and the account has not enabled 18+

Reply:
I assume that if the user wants to see +18 content he will have enabled it before in its account preferences.
I dont think its up to us to modify user preferences. And a warning everytime would be a hassle.

Reply:
>  Shouldn't there be an error for the user when accessing such media and the account has not enabled 18+

Well in that case, FetchPages just return no images as website redirect to main page 🤷

Reply:
Nevermind :)